pykafka/cluster.py

```diff
@@ -234,10 +238,14 @@ def _request_metadata(self, broker_connects, topics):
                                     self._offsets_channel_socket_timeout_ms,
                                     buffer_size=1024 * 1024,
                                     source_host=self._source_host,
-                                    source_port=self._source_port)
+                                    source_port=self._source_port,
+                                    ssl_config=self._ssl_config)
                     response = broker.request_metadata(topics)
                     if response is not None:
                         return response
+                # FIXME: handle ssl errors separately somehow, because if we
+                # don't, _get_metadata() might attempt to treat it as if it
+                # were a zookeeper host, which gets messy
```

Comment:
You can see it when you connect to an SSL port while not providing an `ssl_config` to `KafkaClient`: the connection keeps getting closed (because the broker is expecting some sort of SSL handshake and not getting it), and kazoo stubbornly keeps trying to reconnect - it doesn't figure out that the other end is simply not a Zookeeper machine.